client/src/app/users/user-list.component.spec.ts

```diff
@@ -83,10 +108,17 @@ describe('User list', () => {
   });
 });
 
+/*
+ * This test is a little odd, but illustrates how we can use stubs
+ * to create mock objects (a service in this case) that be used for
+ * testing. Here we set up the mock UserService (userServiceStub) so that
+ * _always_ fails (throws an exception) when you request a set of users.
+ *
+ * So this doesn't really test anything meaningful in the context of our
+ * code (I certainly wouldn't copy it), but it does illustrate some nice
+ * testing tools. Hopefully it's useful as an example in that regard.
+ */
```

Comment:
Oh, I'm not sure if I agree with that. This test case _does_ test something meaningful—namely, what happens to the component if we can't talk to the server? It's useful to know that `userList.serverFilteredUsers` is undefined if that happens, rather than having some garbage value.

Reply:
I'll buy that, and just removed that second paragraph.